src/main/fc/rc.c

```diff
@@ -341,71 +341,65 @@ static FAST_CODE_NOINLINE void rcSmoothingSetFilterCutoffs(rcSmoothingFilter_t *
-        const float setpointCutoffFrequency = smoothingData->setpointCutoffFrequency;
-        for (int i = 0; i < PRIMARY_CHANNEL_COUNT; i++) {
-            if (i < THROTTLE) {
-                if (!smoothingData->filterInitialized) {
-                    pt3FilterInit(&smoothingData->filterSetpoint[i], pt3FilterGain(setpointCutoffFrequency, dT));
-                } else {
-                    pt3FilterUpdateCutoff(&smoothingData->filterSetpoint[i], pt3FilterGain(setpointCutoffFrequency, dT));
-                }
-            } else {
-                const float throttleCutoffFrequency = smoothingData->throttleCutoffFrequency;
-                if (!smoothingData->filterInitialized) {
-                    pt3FilterInit(&smoothingData->filterSetpoint[i], pt3FilterGain(throttleCutoffFrequency, dT));
-                } else {
-                    pt3FilterUpdateCutoff(&smoothingData->filterSetpoint[i], pt3FilterGain(throttleCutoffFrequency, dT));
-                }
-            }
+    const float setpointCutoffFrequency = smoothingData->setpointCutoffFrequency;
+    const float throttleCutoffFrequency = smoothingData->throttleCutoffFrequency;
+
+    if (!smoothingData->filterInitialized) {
+        for (int i = FD_ROLL; i <= FD_YAW; i++) {
+            pt3FilterInit(&smoothingData->filterSetpoint[i], pt3FilterGain(setpointCutoffFrequency, dT));
+            pt3FilterInit(&smoothingData->filterFeedforward[i], pt3FilterGain(setpointCutoffFrequency, dT));
         }
-        // initialize or update the RC Deflection filter
         for (int i = FD_ROLL; i < FD_YAW; i++) {
-            if (!smoothingData->filterInitialized) {
-                pt3FilterInit(&smoothingData->filterRcDeflection[i], pt3FilterGain(setpointCutoffFrequency, dT));
-            } else {
-                pt3FilterUpdateCutoff(&smoothingData->filterRcDeflection[i], pt3FilterGain(setpointCutoffFrequency, dT));
-            }
+           pt3FilterInit(&smoothingData->filterRcDeflection[i], pt3FilterGain(setpointCutoffFrequency, dT));
         }
-    }
-    // initialize or update the Feedforward filter
-    if ((smoothingData->feedforwardCutoffFrequency != oldFeedforwardCutoff) || !smoothingData->filterInitialized) {
-       for (int i = FD_ROLL; i <= FD_YAW; i++) {
-            const float feedforwardCutoffFrequency = smoothingData->feedforwardCutoffFrequency;
-            if (!smoothingData->filterInitialized) {
-                pt3FilterInit(&smoothingData->filterFeedforward[i], pt3FilterGain(feedforwardCutoffFrequency, dT));
-            } else {
-                pt3FilterUpdateCutoff(&smoothingData->filterFeedforward[i], pt3FilterGain(feedforwardCutoffFrequency, dT));
-            }
+        pt3FilterUpdateCutoff(&smoothingData->filterSetpoint[3], pt3FilterGain(throttleCutoffFrequency, dT));
+    }
+
+    // note that cutoff frequencies are integers, filter cutoffs won't re-calculate until there is > 1hz variation from previous cutoff
```

Comment:
Isn't it better to use `float` and explicit delta?  
float value can be close to inteder, so small noise will cause frequent recalculation - nondeterministic additional CPU load can have consequences ...

Reply:
I'm fine making it a float And giving it a 1hz range for recalculating. I'm fine even recalculating every loop and not checking.

Reply:
Personally I'd use each cycle recalc - it's simpler and more deterministic.

Reply:
Perfect, done and done, more code to delete that way anyhow :)

Reply:
@ctzsnooze check out this conversation